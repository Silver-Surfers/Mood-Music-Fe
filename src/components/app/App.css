.app {
  margin: 0;
  min-height: 100vh;
  background-color: #1DB954;
}

* {
  margin: 0;
  padding: 0;
  
}
