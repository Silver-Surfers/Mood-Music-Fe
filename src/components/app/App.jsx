import React from 'react';
import { 
  BrowserRouter as Router, 
  Route, 
  Switch 
} from 'react-router-dom';
import { AuthPage } from '../AuthPage';
import { StillShot } from '../WebCam/StillShot';
import { WebcamCapture } from '../WebCam/WebcamComponent';
import { MusicPlayer } from '../MusicPlayer';
import styles from './App.css';



export default function App() {
  return (
    <div className={styles.app}>
      <Router>
        <Switch>
          <Route 
            exact path="/" 
            component={AuthPage} 
          />
          <Route
            exact path="/webcam"
            component={WebcamCapture}
          />
          <Route 
            exact path="/image" 
            component={StillShot} 
          />
          <Route
            exact path="/media"
            component={MusicPlayer}
          />
        </Switch>
      </Router>
    </div>
  );
}
