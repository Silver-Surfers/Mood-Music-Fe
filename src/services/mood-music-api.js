/* eslint-disable max-len */

export const makeRequest = async(body) => {
  try { const res = await fetch(
    'https://westus.api.cognitive.microsoft.com/face/v1.0/detect?returnFaceId=true&returnFaceLandmarks=false&returnFaceAttributes=emotion&recognitionModel=recognition_03&returnRecognitionModel=false', 
    {
      method: 'POST',
      headers: {
        'Content-Type': 'application/octet-stream',
        'Ocp-Apim-Subscription-Key': `${process.env.AZURE_KEY}`
      },
      body
    });

  const json = await res.json();
  const emotion = json[0].faceAttributes.emotion;
  return emotion;

  } catch(error) {
    console.log(error);
  }
};

export const requestMusic = (emotion, accessToken) => {
  try { return fetch(
    `https://api.spotify.com./v1/search?q=${emotion}&type=playlist`,
    { 
      method: 'GET',
      headers: {
        'Authorization': `Bearer ${accessToken}`,
        'Content-Type' : 'application/json'
      },
    })
  
    .then(res => res.json())
    .then(json => json.playlists.items)
    .then(items => items.map(item => (
      item.id
    )));
  
  } catch(error) {
    console.log(error);
  }
};

// export const requestFollowPlaylist = async() => {
//   try { return fetch(`https://api.spotify.com/v1/playlists/${playlist}/followers`,
//     {
//       method: 'PUT',
//       headers: {
//         'Authorization': `Bearer ${accessToken}`,
//         'Content-Type' : 'application/json'
//       },
//       body: {
//         'public': false
//       }
//     })
//     .then(res => res.json())
//     .then(json => json.playlists.requestFollowPlaylist)
//     .then(requestFollowPlaylist => (
//       requestFollowPlaylist.id
//     ));
//   } catch(error) {
//     console.log(error);
//   }

// };
