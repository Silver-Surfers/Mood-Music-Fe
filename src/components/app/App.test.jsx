/* eslint-disable max-len */
import React from 'react';
import { render, cleanup, fireEvent, waitFor, screen } from '@testing-library/react';
import { createStore } from 'redux';
import reducer from '../../reducers/azureReducer';
import App from './App';
import { makeRequest } from '../../services/mood-music-api';
import { Provider } from 'react-redux';


describe('App component', () => {
  afterEach(() => cleanup());
  it('renders App', async() => {
    // makeRequest([
    //   {    
    //     'anger': '0',
    //     'contempt': '0',
    //     'disgust': '0',
    //     'fear': '0',
    //     'happiness': '0.001',
    //     'neutral': '0.993',
    //     'sadness': '0.002',
    //     'surprise': '0.005'
    //   }
    // ]);

    const store = createStore(reducer);
    const { asFragment } = render(
      <Provider store={store}>
        <App />
      </Provider>
    );

    const button = screen.getByRole('button');

    fireEvent.click(button);
    return waitFor(() => {
      expect(asFragment()).toMatchSnapshot();
    });

  });
});

