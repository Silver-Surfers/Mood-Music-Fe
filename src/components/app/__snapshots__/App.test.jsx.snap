// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component renders App 1`] = `
<DocumentFragment>
  <h1>
    WELCOME TO MOOD MUSIC
  </h1>
  <button>
    Login To Spotify
  </button>
</DocumentFragment>
`;
